cmd/elastic_ip_create.go

```diff
@@ -53,56 +49,70 @@ func (c *elasticIPCreateCmd) cmdPreRun(cmd *cobra.Command, args []string) error
+	ctx := gContext
+	client, err := switchClientZoneV3(ctx, globalstate.EgoscaleV3Client, c.Zone)
+	if err != nil {
+		return err
+	}
 
-	var healthcheck *egoscale.ElasticIPHealthcheck
+	var healthcheck *v3.ElasticIPHealthcheck
 	if c.HealthcheckMode != "" {
-		port := uint16(c.HealthcheckPort)
-		interval := time.Duration(c.HealthcheckInterval) * time.Second
-		timeout := time.Duration(c.HealthcheckTimeout) * time.Second
-
-		healthcheck = &egoscale.ElasticIPHealthcheck{
-			Interval:    &interval,
-			Mode:        &c.HealthcheckMode,
-			Port:        &port,
-			StrikesFail: &c.HealthcheckStrikesFail,
-			StrikesOK:   &c.HealthcheckStrikesOK,
-			Timeout:     &timeout,
-			URI: func() (v *string) {
+		healthcheck = &v3.ElasticIPHealthcheck{
+			Interval:    c.HealthcheckInterval,
+			Mode:        v3.ElasticIPHealthcheckMode(c.HealthcheckMode),
+			Port:        c.HealthcheckPort,
+			StrikesFail: c.HealthcheckStrikesFail,
+			StrikesOk:   c.HealthcheckStrikesOK,
+			Timeout:     c.HealthcheckTimeout,
+			URI: func() (v string) {
 				if strings.HasPrefix(c.HealthcheckMode, "http") {
-					v = &c.HealthcheckURI
+					v = c.HealthcheckURI
 				}
 				return
 			}(),
 		}
 
 		if c.HealthcheckMode == "https" {
-			healthcheck.TLSSkipVerify = &c.HealthcheckTLSSSkipVerify
-			healthcheck.TLSSNI = utils.NonEmptyStringPtr(c.HealthcheckTLSSNI)
+			healthcheck.TlsSkipVerify = &c.HealthcheckTLSSSkipVerify
+			healthcheck.TlsSNI = c.HealthcheckTLSSNI
 		}
 	}
 
-	elasticIP := &egoscale.ElasticIP{
-		Description: utils.NonEmptyStringPtr(c.Description),
+	createElasticIPRequest := v3.CreateElasticIPRequest{
+		Description: c.Description,
 		Healthcheck: healthcheck,
 	}
 
 	if c.IPv6 {
-		elasticIP.AddressFamily = utils.NonEmptyStringPtr("inet6")
+		createElasticIPRequest.Addressfamily = v3.CreateElasticIPRequestAddressfamilyInet6
 	}
 
-	var err error
-	decorateAsyncOperation("Creating Elastic IP...", func() {
-		elasticIP, err = globalstate.EgoscaleClient.CreateElasticIP(ctx, c.Zone, elasticIP)
+	err = decorateAsyncOperations("Creating Elastic IP...", func() error {
+		op, err := client.CreateElasticIP(ctx, createElasticIPRequest)
+		if err != nil {
+			return fmt.Errorf("exoscale: error while creating Elastic IP: %w", err)
+		}
+
+		_, err = client.Wait(ctx, op, v3.OperationStateSuccess)
+		if err != nil {
+			return fmt.Errorf("exoscale: error while waiting for Elastic IP creation: %w", err)
+		}
+
+		return nil
 	})
+
 	if err != nil {
 		return err
 	}
 
 	return (&elasticIPShowCmd{
 		cliCommandSettings: c.cliCommandSettings,
-		ElasticIP:          *elasticIP.ID,
-		Zone:               c.Zone,
+		// Comment for reviewer:
+		// Is there a way to get the created Elastic IP address or UUID ???
+		// Listing all of them and comparing Addressfamily, Description,... doesn't really garantee uniquess
+		// TODO: Remove comment before merging
+		ElasticIP: "",
+		Zone:      c.Zone,
```

Comment:
yes you need to get the `op` after waiting and there is the `op.Reference.ID` check if reference not nil before using

Reply:
Is there a way to get the `op` from the `decorateAsyncOperations` ?? 
As this function is only returning an error if it happens ??!!